File: resource-registry/src/test/java/nl/haploid/octowight/registry/repository/ResourceRootDmoRepositoryIT.java
package nl.haploid.octowight.registry.repository;

import nl.haploid.octowight.registry.AbstractIT;
import nl.haploid.octowight.registry.TestData;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Collections;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class ResourceRootDmoRepositoryIT extends AbstractIT {

	@Autowired
	private ResourceRootDmoRepository resourceRootDmoRepository;

	@Test
	public void findByAtomIdAndAtomTypeAndAtomLocus() {
		resourceRootDmoRepository.deleteAll();
		final ResourceRootDmo dmo = TestData.resourceRootDmo();
		final ResourceRootDmo expectedDmo = resourceRootDmoRepository.save(dmo);
		final ResourceRootDmo actualDmo = resourceRootDmoRepository
				.findByResourceTypeAndAtomIdAndAtomTypeAndAtomLocus(expectedDmo.getResourceType(),
						expectedDmo.getAtomId(), expectedDmo.getAtomType(), expectedDmo.getAtomLocus());
		assertEquals(expectedDmo, actualDmo);
	}

	@Test
	public void findByResourceType() {
		resourceRootDmoRepository.deleteAll();
		final ResourceRootDmo dmo1 = TestData.resourceRootDmo("willow");
		final ResourceRootDmo dmo2 = TestData.resourceRootDmo("oz");
		final ResourceRootDmo expectedDmo1 = resourceRootDmoRepository.save(dmo1);
		resourceRootDmoRepository.save(dmo2);
		final List<ResourceRootDmo> actualDmos = resourceRootDmoRepository.findByResourceType("willow");
		final List<ResourceRootDmo> expectedDmos = Collections.singletonList(expectedDmo1);
		assertEquals(expectedDmos, actualDmos);
	}

	@Test
	public void findByResourceTypeAndResourceId() {
		resourceRootDmoRepository.deleteAll();
		final ResourceRootDmo dmo1 = TestData.resourceRootDmo("willow");
		final ResourceRootDmo dmo2 = TestData.resourceRootDmo("oz");
		final ResourceRootDmo expectedDmo = resourceRootDmoRepository.save(dmo1);
		resourceRootDmoRepository.save(dmo2);
		final ResourceRootDmo actualDmo = resourceRootDmoRepository.findByResourceTypeAndResourceId("willow", expectedDmo.getResourceId());
		assertEquals(expectedDmo, actualDmo);
	}
}

File: resource-registry/src/test/scala/nl/haploid/octowight/registry/repository/ResourceRootDmoRepositoryIT.scala
package nl.haploid.octowight.registry.repository

import java.util

import nl.haploid.octowight.registry.{AbstractIT, TestData}
import org.springframework.beans.factory.annotation.Autowired

class ResourceRootDmoRepositoryIT extends AbstractIT {
  @Autowired private[this] val resourceRootDmoRepository: ResourceRootDmoRepository = null

  behavior of "Resource root repository"

  it should "find by atom id, type & origin" in {
    resourceRootDmoRepository.deleteAll()
    val dmo = TestData.resourceRootDmo
    val expectedDmo = resourceRootDmoRepository.save(dmo)
    val actualDmo = resourceRootDmoRepository.findByResourceCollectionAndRootIdAndRootCategoryAndRootOrigin(
      expectedDmo.resourceCollection, expectedDmo.root.id, expectedDmo.root.category, expectedDmo.root.origin)
    actualDmo should be(expectedDmo)
    println(actualDmo)
  }

  it should "find by resource collection" in {
    resourceRootDmoRepository.deleteAll()
    val dmo1 = TestData.resourceRootDmo("willow")
    val dmo2 = TestData.resourceRootDmo("oz")
    val expectedDmo1 = resourceRootDmoRepository.save(dmo1)
    resourceRootDmoRepository.save(dmo2)
    val actualDmos = resourceRootDmoRepository.findByResourceCollection("willow")
    val expectedDmos = util.Collections.singletonList(expectedDmo1)
    actualDmos should be(expectedDmos)
  }

  it should "find by resource collection & id" in {
    resourceRootDmoRepository.deleteAll()
    val dmo1 = TestData.resourceRootDmo("willow")
    val dmo2 = TestData.resourceRootDmo("oz")
    val expectedDmo = resourceRootDmoRepository.save(dmo1)
    resourceRootDmoRepository.save(dmo2)
    val actualDmo = resourceRootDmoRepository.findByResourceCollectionAndResourceId("willow", expectedDmo.resourceId)
    actualDmo should be(expectedDmo)
  }
}
